pkg/cmd/openshift-operators/apiserver-operator/apis/apiserver/v1/types.go

```diff
@@ -0,0 +1,48 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type OpenShiftAPIServerConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   OpenShiftAPIServerConfigSpec   `json:"spec"`
+	Status OpenShiftAPIServerConfigStatus `json:"status"`
+}
+
+type OpenShiftAPIServerConfigSpec struct {
+	// TODO I think this should eventually embed the entire master config
+	// it will end up overlaying in the following order:
+	// 1. hardcoded default
```

Comment:
This merging logic here is another core principle that doesn't exist upstream since they don't use serialized config.  We need something to apply per-version changes and give a gentle onramp.  The defaults hardcoded into the operator do that.  The existing config overlaying ensures that we don't remove values.  Applying this config last allows complete control if you desire it.

I could envision a "4" based on the logical status values.  Things like "this is the service serving cert CA" because that is a status value from the SSCS controller.